include/class.search.php

```diff
@@ -1399,8 +1406,14 @@ class DepartmentManagerSelectionField extends AgentSelectionField {
     static $_members;
 
     function getChoices($verbose=false) {
-        if (isset($this->_members))
-            $this->_members = Staff::getStaffMembers();
+        if (!isset($this->_members)) {
+            $managers = array();
+            foreach (Staff::objects() as $staff) {
```

Comment:
Why not filter via ORM and select only managers?

Reply:
Off of departments?

```
if (!isset($this->_members)) {
            $managers = array();
            $managerIds = Dept::objects()
                ->filter(Q::any(array(
                    'manager_id__gt' => 0,
                )))
                ->values_flat('manager_id');

            foreach ($managerIds as $key => $id) {
                $staff = Staff::lookup($id[0]);
                $managers['s'.$id[0]] = $staff->getName()->name;
            }
            $this->_members = $managers;
        }
```

Does this make it less expensive? I still ended up with a lookup on the staff member.